develop/vagrant/README.md

```diff
@@ -21,24 +34,181 @@ It is recommended to then install the following VSCode extensions if you don't a
+```
+cd srcDsm/DSM/src
+mkdir build
+cd build
+  * build without rbus
+    cmake ../..
+* build  rbus
+    cmake ../.. -D ENABLE_RBUS_PROVIDER=ON
+make
+sudo make install
+```
+
+## 9. Building Dobby inside vagrant.
+```
+cd ~/srcDobby/build
+cmake ../
 make
 sudo make install
 ```
 
-It is possible to add build-time arguments to CMake when using the "Build" button in VSCode (e.g. to make sure the platform is set to DEV_VM). Open/create the file .vscode/settings.
-json and add the following JSON section:
-```json
-"cmake.configureArgs": [
-    "-DRDK_PLATFORM=DEV_VM"
-]
+### 11. Setting up a server with python to host containers
 ```
+sudo python3 -m http.server 8080
+
+```
+
+### 12. Clean up any installed containers.
+```
+rm -rf ~/destination/*
+
+```
+
+
+### 10. Run inside vagrant DSM with Dobby and rbus
+```
+vagrant ssh
+sudo DobbyDaemon --nofork
+with rbus we also need
+vagrant ssh
+killall -9 rtrouted;rm -fr /tmp/rtroute*;rtrouted -f -l DEBUG
+```
+In he next terminal start dsm:
+```
+vagrant ssh
+dsm
+
+```
+
+
+### 11. Deployment containers via rbus:
+
+* DSM rbus provider implements installDU() with URL (mandatory) and name (optional default is "default")
+
+
+* build and install DSM
+* start rtrouted
+* start DSM ( sudo DSM )
+* rbus installDU gets called via
+
+```
+rbuscli method_values "Device.SoftwareModules.InstallDU()" URL string http://${SERVER_IP}:8080/example_container.tar 
+```
+ExecutionEnvRef string test
+* check state of example_container via
+
+Check container gets deployed .
+ls -al ~destination/
+
+```
+rbuscli getvalues "Device.SoftwareModules.ExecutionUnit."
+```
+  It will show status Idle.
+* start example_container container dsmcli eu.start <eu_id>
+* check state of example_container via
+
+```
+rbuscli getvalues "Device.SoftwareModules.ExecutionUnit."
+```
+  It will show status Active.
+* dsmcli eu.detail <eu_id> will show status Active.
+* sudo DobbyTool list will show container example_container running.
+* Stop example_container container dsmcli eu.stop <eu_id>
+* check state of example_container via rbuscli getvalues "Device.SoftwareModules.ExecutionUnit." It will show status Idle
+
+
+### 12. Set EU Active and Idle state via rbus:
+
+DSM's functions start/stop available over RBUS SetRequestedState
+```
+rbuscli method_values "Device.SoftwareModules.ExecutionUnit.1.SetRequestedState()" RequestedState string Active
+#above line is equivalent to dsmcli eu.start <eu_id>
+rbuscli method_values "Device.SoftwareModules.ExecutionUnit.1.SetRequestedState()" RequestedState string Idle
+#above line is equivalent to dsmcli eu.stop <eu_id>
+```
+
+### 13. DSM du.uninstall available over RBUS as well.
+
+Demo
+
+
+```
+
+Check via rbus
+
+rbuscli getvalues "Device.SoftwareModules.DeploymentUnit."
+
+uninstalled via rbus
+rbuscli method_noargs "Device.SoftwareModules.DeploymentUnit.1.Uninstall()"
+
+Check , It will show empty.
+rbuscli getvalues "Device.SoftwareModules.DeploymentUnit."
```

Comment:
Comments on non-commands so a developer can copy paste these instructions straight into the terminal
```suggestion

#Check via rbus

rbuscli getvalues "Device.SoftwareModules.DeploymentUnit."

#uninstalled via rbus
rbuscli method_noargs "Device.SoftwareModules.DeploymentUnit.1.Uninstall()"

#Check , It will show empty.
rbuscli getvalues "Device.SoftwareModules.DeploymentUnit."
```

Reply:
Addressed.